tests/test_etherscan.py

```diff
@@ -1,21 +1,63 @@
 import json
+from pathlib import Path
 
 import pytest
 from ape import networks
 from requests import Response
 
+ADDRESS = "0xab5801a7d398351b8be11c439e05c5b3259aec9b"
+TRANSACITON = "0x0da22730986e96aaaf5cedd5082fea9fd82269e41b0ee020d966aa9de491d2e6"
+
+
+@pytest.fixture
+def explorer():
+    return networks.ethereum.mainnet.explorer
+
 
 @pytest.fixture
 def etherscan_abi_response(mocker):
     response = mocker.MagicMock(spec=Response)
-    f = open("tests/get_contract_response.json")
-    response.json.return_value = json.load(f)
-    f.close()
-    return response
+    test_data_path = Path(__file__).parent / "get_contract_response.json"
+    with open(test_data_path) as response_data_file:
+        response.json.return_value = json.load(response_data_file)
+        yield response
+
+
+def setup_mock_get(mocker, etherscan_abi_response, expected_params):
+    get_patch = mocker.patch("ape_etherscan.client.requests")
+
+    def get(base_uri, params=None, *args, **kwargs):
+        # Request will fail if made with incorrect parameters.
+        assert base_uri == "https://api.etherscan.io/api"
+        assert params == expected_params
+        return etherscan_abi_response
+
+    get_patch.get.side_effect = get
+    return get_patch
+
+
+def test_get_address_url(explorer):
```

Comment:
Can we turn these into `@pytest.mark.parametrize` cases (with one entry to start) so it's easy to add new cases to try for?

Also when this plugin supports explorers for multiple networks (in fact should probably try it with all the eth testnets)

This and the transaction one and contract type one

Reply:
also maybe try different types contracts at different addresses (proxies, no contract at address, unpublished, etc.)

Reply:
I can definitely add tests for the test nets.
I don't really know how different contract types with affect the outcome on our side... we just return a str address?

re: the `parametrize()`, I am not exactly sure what you mean! What is being parametrized here?

Reply:
Oh whoops, I mean for get contract type

parametrize the networks (and addresses, if relevant)

Reply:
Oh, would the ABI serialization stuff differ in this case? Re: the different types of contracts

Reply:
Yes, it might!

Reply:
Okay I parametrized the network, the expected URL, and the address / txn has (testing one of the testnets too).

I will admit I am not a huge fan of parametrizing tests to this level, but that is just my personal preference.

Testing the contract types stuff will take some more research on my end.